src/main/java/SequenceProvider.java

```diff
@@ -51,4 +51,80 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
     }
 
+    public String fillAllMissingInfo(String sequenceJsonString) throws IOException{
+
+        MongoClient mongoClient = Util.getMongoClient();
+        MongoDatabase db = mongoClient.getDatabase(Util.DB_NAME);
+        MongoCollection collection = db.getCollection(Util.COURSE_DATA_COLLECTION_NAME);
+        JSONObject sequenceJson = new JSONObject();
+        try{
+
+            // convert to json object
+            sequenceJson = new JSONObject(sequenceJsonString);
+            JSONArray semesterList = sequenceJson.getJSONArray("semesterList");
+
+            // traverse through, filling missing info on each valid course code
+            for(int i = 0; i < semesterList.length(); i++){
+                JSONObject semester = semesterList.getJSONObject(i);
+                JSONArray courseList = semester.getJSONArray("courseList");
+                for(int j = 0; j < courseList.length(); j++){
+                    Object entry = courseList.get(j);
+                    if(entry instanceof JSONObject){
+                        courseList.put(j, fillMissingInfo((JSONObject) entry, collection));
+                    } else if(entry instanceof JSONArray) {
+                        JSONArray orList = (JSONArray) entry;
+                        for(int k = 0; k < orList.length(); k++){
+                            orList.put(k, fillMissingInfo((JSONObject) orList.get(k), collection));
+                        }
+                        courseList.put(j, orList);
+                    } else {
+                        logger.warn("Found unusual value inside course sequence. Expected a course object but found: " + entry.toString());
+                    }
+                }
+                semester.put("courseList", courseList);
+                semesterList.put(i, semester);
+            }
+            sequenceJson.put("semesterList", semesterList);
+
+        } catch(JSONException e){
+            e.printStackTrace();
+            throw new IOException("Error parsing sequence JSON data : " + sequenceJsonString);
+        }
+
+        return sequenceJson.toString();
+    }
+
+    // add in name and credits values from the DB
```

Comment:
I guess we'll do prereqs/coreqs when that feature is implemented. I like that you took initiative to implement something even though this can't be the final version of this function.  Gotta do that more myself.